### ISSUE TITLE
Inconsistent search results when searching for stickers by search terms

### ISSUE THREAD

**Issue by Timbobeek** [signalapp/Signal-Android#11482]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When searching for stickers using search terms, the search seems inconsistent. For example, entering one letter of a supposed tag word loads one set of stickers, but adding another letter loads a completely different set of stickers.

### Steps to reproduce
-Open Signal
-Open any chat
-Tap on "sticker" option
-Tap on "search"
-Enter different symbols ("s", "sa", "sad"), or a similar one, e.g. ("d", "dr", "dro", "drop")

**Actual result:** 
- "s" finds 2 stickers
- "sa" finds 3 completely different stickers
- "sad" adds extra sticker to the previous 3

**Expected result:** 
One would expect that "s", "sa", "sad" will gradually filter out results of the previous term

### Screenshots

https://user-images.githubusercontent.com/68671029/126711772-01a5b618-c0bc-441f-aad0-a57ac51f73e5.mp4

https://user-images.githubusercontent.com/68671029/126711799-868a5061-b761-4676-9198-dca49c377296.mp4

### Device info

**Device:** Samsung Galaxy S6 Edge
**Android version:** 7.0
**Signal version:** 5.15.6

### Link to debug log

https://debuglogs.org/35d8b2fd946223f205e82bc8964de6c5a6cdc04d4df72e55f143463134ec9eec

**Comment by greyson-signal** [comment]
Sticker packs don't have search terms associated with them. But they *do* have emoji associated with them, which we do have search terms for. So sticker search is really emoji search, which we then use to find stickers :) A little roundabout, but it does let you search for things like "love" or "smile" and get back relevant results.

When you search just the letter 's', you get back *a lot* of emoji. We can't search stickers with all of them, so we limit it to 20. It may be that we pick the wrong 20, and don't get back a lot of stickers. Then, when you add letters ('sa', 'sad') we are more likely to get relevant emoji to search for, so the results get better.

So basically you're seeing results because of the 'emoji overflow' problem. Definitely room for improvement here, but largely in the space of improving our results for emoji search and/or the performance of being able to lookup stickers by emoji.

I don't expect to spend much time on this in the immediate future, but improving emoji search by taking into account usage frequency or something sounds like a neat project.

**Comment by Timbobeek** [comment]
@greyson-signal Thank you, makes sense!
